### PR TITLE
removed MklDnn dependency from C#

### PR DESCRIPTION
**Pull request by shahasad**

### CODE REVIEW

**Comment by snnn** [comment]
I'm ok with this change, but I must say, in many many cases, mkldnn increases our performance a lot.

**Comment by yufenglee** [comment]
We can use MLAS by default. And leave users an option to register MKLDNN.

**Comment by snnn** [comment]
> We can use MLAS by default. And leave users an option to register MKLDNN.

Can we kick out all the mkldnn code from the cpu execution provider?

File: csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs [hunk 1]
```diff
@@ -46,10 +46,9 @@ public static SessionOptions Default
             }
         }
 
-        private static SessionOptions MakeSessionOptionWithMklDnnProvider()
```
Comment (yufenglee) [inline csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs]: Maybe we can make sessionoption register cpu provider on construction. And make SessionOptions only have 2 methods: RegisterCPUProvider and RegisterCudaProvider.

Comment (shahasad) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs]: the plan is to package the binary built without the --use_mkldnn option. Thus, providing the user option to register a provider that is not available seems problematic.

**Comment by pranavsharma** [comment]
LGTM 👍

**Comment by pranavsharma** [comment]
> > We can use MLAS by default. And leave users an option to register MKLDNN.
> 
> Can we kick out all the mkldnn code from the cpu execution provider?

only math_cpu.cc has mkldnn function invocations. i can see only mkldnn_sgemm. is there any other place you see mkldnn code in the cpu exec provider?

**Comment by snnn** [comment]
> > > We can use MLAS by default. And leave users an option to register MKLDNN.
> > 
> > 
> > Can we kick out all the mkldnn code from the cpu execution provider?
> 
> only math_cpu.cc has mkldnn function invocations. i can see only mkldnn_sgemm. is there any other place you see mkldnn code in the cpu exec provider?

Can we remove them out of that file?
Now we have 3 different gemm implementations in CPU execution provider. We are not able to test all of them, unless increasing building time by 3 times.
If a kernel really wants to get benefit from mkldnn, we can copy it to mkldnn provider?

File: csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs (hunk 1 above)
Comment (yufenglee) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs]: the it makes sense to remove mkldnn.

Comment (shahasad) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs]: BTW, I will look into your refactoring suggestion as a separate work item.

**Comment by JiguangShen** [comment]
My network inference speed is 8 times slower with MLAS compared to previous version with MKLDNN. 
Can you add some examples to register mkldnn?

**Pull request merged**